thirdparty/netvlad/netvlad.py

```diff
@@ -0,0 +1,186 @@
+"""Source: Adapted from "Hierarchical-Localization" toolbox.
```

Comment:
I worry that if we use all submodules, we'll start to have a pretty heavyweight set of dependencies, since some of these libraries are big (for example, hloc has many submodules itself). But I'm open to the idea in the future, if we feel like the current solution isn't working.